src/main/webapp/pages/dataexchange/studyElectronicOrderView.jsp

```diff
@@ -0,0 +1,388 @@
+<%@page import="org.openelisglobal.internationalization.MessageUtil"%>
+<%@ page language="java" contentType="text/html; charset=UTF-8"%>
+
+<%@ page isELIgnored="false" %>
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form"%>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags"%>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core"%>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="ajax" uri="/tags/ajaxtags" %> 
+<%@ taglib prefix="fn" uri="http://java.sun.com/jsp/jstl/functions" %>
+
+<link href="select2/css/select2.min.css" rel="stylesheet" />
+<script src="select2/js/select2.min.js"></script>
+<script type="text/javascript" src="scripts/tableSort.js"></script>
+<script type="text/javascript" src="scripts/utilities.js"></script>
+<script type="text/javascript" src="scripts/ajaxCalls.js"></script>
+<script type="text/javascript" src="scripts/jquery-ui.js?"></script>
+<script type="text/javascript">
+
+var colSort = [0,0,0,0,0,0,0,0,0,0,0,0];
+var dirty = false;
+var entering = false;
+
+function makeDirty() {
+    dirty = true;
+    function formWarning() {
+        return "<spring:message code="banner.menu.dataLossWarning"/>";
+    }
+    window.onbeforeunload = formWarning;
+}
+
+function searchByIdentifier() {
+	const params = new URLSearchParams({
+		searchType: "IDENTIFIER",
+		searchValue: jQuery('#searchValue').val(),
+		useAllInfo: jQuery('#allInfo1').is(':checked'),
+		});
+	
+	window.location.href = "StudyElectronicOrders?" + params.toString();
+
+}
+
+function searchByFacility() {
+	const params = new URLSearchParams({
+		searchType: "FACILITY",
+		organizationId: jQuery('#organizationId').val(),
+		useAllInfo: jQuery('#allInfo2').is(':checked'),
+		});
+	
+	window.location.href = "StudyElectronicOrders?" + params.toString();
+
+}
+
+function searchByDateAndStatus() {
+	const params = new URLSearchParams({
+		searchType: "DATE_STATUS",
+		startDate: jQuery('#startDate').val(),
+		endDate: jQuery('#endDate').val(),
+		statusId: jQuery('#statusId').val(),
+		useAllInfo: jQuery('#allInfo3').is(':checked'),
+		});
+	
+	window.location.href = "StudyElectronicOrders?" + params.toString();
+
+}
+
+function sort(col) {
+	if (entering) {
+		alert('No sorting while data is being entered');
+	} else {
+		sortTableJquery('eOrderTable', col, ((colSort[col - 1]++) % 2) != 0);
+	}
+}
+
+
+function enterOrder(index) {
```

Comment:
this function has a few XSS vulnerabilities. Please escape jstl for a javascript context ie `<spring:escapeBody javaScriptEscape="true">${labPrefix}</spring:escapeBody>`

it is also recommended to use DOM creation for all object creation so that xss cannot occur throught html string concatenation